packages/teleport/src/components/QuickLaunch/QuickLaunch.jsx

```diff
@@ -63,10 +63,14 @@ export default function FieldInputSsh({
   );
 }
 
-const SSH_STR_REGEX = /(^(\w+-?\w+)+@(\S+)$)/;
+// SSH_STR_REGEX is a modified regex from teleport's lib/sshutils/scp/scp.go.
+// Captures two named groups: username and host.
+const SSH_STR_REGEX = /(?:(?<username>[-.\w@]+)@)(?<host>[-.\w]+)$/;
 const check = value => {
-  const match = SSH_STR_REGEX.exec(value);
-  return match !== null;
+  const hasWhiteSpace = /\s/.test(value);
+  if (!hasWhiteSpace) {
+    return SSH_STR_REGEX.exec(value);
+  }
```

Comment:
I couldn't keep the validation simple because mostly all special characters made the terminal look like it was hanging (but it is a `400` error), I thought it'd be simpler to prevent this error by regex: 
![Screenshot from 2020-07-13 21-15-33](https://user-images.githubusercontent.com/43280172/87389367-7a228580-c55b-11ea-91e1-da7136ba9f88.png)

The regex was modeled after this: https://github.com/gravitational/teleport/blob/master/lib/sshutils/scp/scp.go#L652. @benarent mentioned this PR: https://github.com/gravitational/teleport/pull/4002, something like `root@domain@127.0.0.1` will correctly resolve to

```
username: root@domain
host: 127.0.0.1
```

Dashes and periods seems to be safe and gives correct `disconnect` errors when it can't connect:
![Screenshot from 2020-07-13 22-58-52](https://user-images.githubusercontent.com/43280172/87389860-73e0d900-c55c-11ea-96af-b2702cf2a9ac.png)

Reply:
> Dashes and periods seems to be safe and gives correct disconnect errors when it can't connect:

This is not up to the dashes and input field to control the correctness of the server disconnect errors. Could you check why these errors are not handled correctly on the webapi level?

Reply:
> I couldn't keep the validation simple because mostly all special characters made the terminal look like it was hanging (but it is a 400 error),

Just to clarify, does it eventually show an error? (no such host, etc) ?

Reply:
> Just to clarify, does it eventually show an error? (no such host, etc) ?

No, it just seems to send out a close event right away. I will look into the error

Reply:
note: problem was caused by url's not matching https://github.com/gravitational/webapps/blob/master/packages/teleport/src/console/stores/storeDocs.ts#L84, fixed by encoding URL